catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -86,7 +89,16 @@ public LazyProcessResourceImpl(String metacardId, Supplier<Resource> resourceSup
   @Override
   public InputStream getInputStream() throws IOException {
     loadResource();
-    return inputStream;
+    if (inputStreamDataCache == null) {
```

Comment:
❗️ I do have concurrency concerns here since we are dealing with 2 attributes being updated at different time and also because we are initializing the `inputStreamDataCache` before it is actually usable. I would make this method and `close()` both synchronized